Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -101,38 +103,31 @@ public function handle(CommandInterface $command): void
-        $initiatingUserId = $this->userIdProvider->getUserId();
-        $initiatingTimestamp = $this->clock->now()->format(\DateTimeInterface::ATOM);
-
-        // Add "initiatingUserId" and "initiatingTimestamp" metadata to all events.
-        //                        This is done in order to keep information about the _original_ metadata when an
-        //                        event is re-applied during publishing/rebasing
-        // "initiatingUserId": The identifier of the user that originally triggered this event. This will never
-        //                     be overridden if it is set once.
-        // "initiatingTimestamp": The timestamp of the original event. The "recordedAt" timestamp will always be
-        //                        re-created and reflects the time an event was actually persisted in a stream,
-        // the "initiatingTimestamp" will be kept and is never overridden again.
-        // TODO: cleanup
-        $eventsToPublish = new EventsToPublish(
-            $eventsToPublish->streamName,
-            Events::fromArray(
-                $eventsToPublish->events->map(function (EventInterface|DecoratedEvent $event) use (
-                    $initiatingUserId,
-                    $initiatingTimestamp
-                ) {
-                    $metadata = $event instanceof DecoratedEvent ? $event->eventMetadata?->value ?? [] : [];
-                    $metadata['initiatingUserId'] ??= $initiatingUserId;
-                    $metadata['initiatingTimestamp'] ??= $initiatingTimestamp;
-                    return DecoratedEvent::create($event, metadata: EventMetadata::fromArray($metadata));
-                })
-            ),
-            $eventsToPublish->expectedVersion,
-        );
-
-        $this->eventPersister->publishEvents($this, $eventsToPublish);
+        $eventsToPublishOrGenerator = $this->commandBus->handle($command, $this->commandHandlingDependencies);
+
+        if ($eventsToPublishOrGenerator instanceof EventsToPublish) {
+            $eventsToPublish = $this->enrich($eventsToPublishOrGenerator);
+            $this->eventPersister->publishEvents($this, $eventsToPublish);
+        } else {
+            foreach ($eventsToPublishOrGenerator as $eventsToPublish) {
+                assert($eventsToPublish instanceof EventsToPublish); // just for the ide
+                $eventsToPublish = $this->enrich($eventsToPublish);
+                try {
+                    $this->eventPersister->publishEvents($this, $eventsToPublish);
+                } catch (ConcurrencyException $concurrencyException) {
+                    $errorStrategy = $eventsToPublishOrGenerator->send(new EventsToPublishFailed(
+                        $eventsToPublish->expectedVersion,
+                        $concurrencyException
+                    ));
+
+                    if ($errorStrategy instanceof EventsToPublish) {
+                        $this->eventPersister->publishEvents($this, $errorStrategy);
```

Comment:
should we enrich the error strategy here with the metadata? Or just move the enrichment to the event persister?